arraycontext/impl/pyopencl/__init__.py

```diff
@@ -133,6 +148,10 @@ def from_numpy(self, array: np.ndarray):
         return cl_array.to_device(self.queue, array, allocator=self.allocator)
 
     def to_numpy(self, array):
+        from numbers import Number
+        if not self._force_device_scalars and isinstance(array, Number):
```

Comment:
I think this doesn't work for a `boolean` scalar. Probably `np.isscalar(array)`?

Reply:
Good point! Added in b20ac9a.